ui/src/main/webapp/wizards/sources/index.js

```diff
@@ -23,15 +23,16 @@ const WizardView = ({ id, children, clearWizard }) => (
 
 const Wizard = connect(null, { clearWizard })(WizardView)
 
+const stageMapping = {
+  welcomeStage: WelcomeStage,
+  discoveryStage: DiscoveryStage,
+  sourceSelectionStage: SourceSelectionStage,
+  confirmationStage: ConfirmationStage,
+  completedStage: CompletedStage
+}
+
 let StageRouter = ({ stage, messages }) => {
-  const stageMapping = {
-    welcomeStage: <WelcomeStage />,
-    discoveryStage: <DiscoveryStage />,
-    sourceSelectionStage: <SourceSelectionStage />,
-    confirmationStage: <ConfirmationStage />,
-    completedStage: <CompletedStage />
-  }
-  return (stageMapping[stage])
+  return React.createElement(stageMapping[stage])
```

Comment:
@tbatie  because `<stageMapping[stage] />` is not valid jsx syntax, we have to call `React.createElement` directly.

However, an alternative would be to bind the name first, and then create the jsx element:

```javascript
const Stage = stageMapping[stage]
return <Stage />
```

Either solution works for me.

https://facebook.github.io/react/docs/jsx-in-depth.html#choosing-the-type-at-runtime